newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -186,10 +195,36 @@ export default class AuthenticatedUserProvider extends React.Component<
   _fetchUserProfile = async () => {
     const { authentication } = this.props;
 
+    this.setState(({ authenticatedUser }) => ({
+      authenticatedUser: {
+        ...authenticatedUser,
+        loginState: 'loading',
+      },
+    }));
+
     // First ensure the Firebase authenticated user is up to date
     // (and let the error propagate if any).
-    const firebaseUser = await this._reloadFirebaseProfile();
-    if (!firebaseUser) return;
+    let firebaseUser;
+    try {
+      firebaseUser = await this._reloadFirebaseProfile();
+      if (!firebaseUser) {
+        this.setState(({ authenticatedUser }) => ({
+          authenticatedUser: {
+            ...authenticatedUser,
+            loginState: 'done',
```

Comment:
Maybe a dumb question, but will `loginState: 'done'` be set if there is no error?

Reply:
I'm not sure I see which case you're referring to

Reply:
I was mentioning the "happy path", where `firebaseUser = await this._reloadFirebaseProfile();` works and does not return a null object. But this is handled at the end of the function already :)